### ISSUE TITLE
Detach gradients when evaluation to reduce memory cost

### ISSUE THREAD

**Issue by jytime** [facebookresearch/votenet#30]
Hi Charles,

There is an evaluation process after every 10 epochs.
https://github.com/facebookresearch/votenet/blob/237f6ffd2ba95e7c8667d25e59e328b908d7fefd/train.py#L319-L320

I guess the gradients during evaluation are unnecessary while they take a lot of GPU memory. It may be better to detach gradients during evaluation, just as:

```
 if EPOCH_CNT == 0 or EPOCH_CNT % 10 == 9: # Eval every 10 epochs 
     with torch.no_grad():
          loss = evaluate_one_epoch() 
```
or more specifically, detach gradients in evaluate_one_epoch().
https://github.com/facebookresearch/votenet/blob/237f6ffd2ba95e7c8667d25e59e328b908d7fefd/train.py#L268
```
     with torch.no_grad():
          end_points = net(inputs)
```

In my experiments, it decreases around 2300 MiB memory cost when batch size=8 (the original cost is 8877 MiB).

Best,
Jianyuan Wang

**Comment by charlesq34** [comment]
That's a very good suggestion!

I've updated the `train.py` script accordingly.
